pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -365,14 +365,44 @@ func (cm *containerManagerImpl) NewPodContainerManager() PodContainerManager {
 			enforceCPULimits:  cm.EnforceCPULimits,
 			// cpuCFSQuotaPeriod is in microseconds. NodeConfig.CPUCFSQuotaPeriod is time.Duration (measured in nano seconds).
 			// Convert (cm.CPUCFSQuotaPeriod) [nanoseconds] / time.Microsecond (1000) to get cpuCFSQuotaPeriod in microseconds.
-			cpuCFSQuotaPeriod: uint64(cm.CPUCFSQuotaPeriod / time.Microsecond),
+			cpuCFSQuotaPeriod:   uint64(cm.CPUCFSQuotaPeriod / time.Microsecond),
+			podContainerManager: cm,
 		}
 	}
 	return &podContainerManagerNoop{
 		cgroupRoot: cm.cgroupRoot,
 	}
 }
 
+func (cm *containerManagerImpl) PodHasExclusiveCPUs(pod *v1.Pod) bool {
+	for _, container := range pod.Spec.InitContainers {
+		exclusiveCPUs := cm.cpuManager.GetExclusiveCPUs(string(pod.UID), container.Name)
+		if !exclusiveCPUs.IsEmpty() {
```

Comment:
I'm still inclined to `return true` if `!exclusiveCPUs.IsEmpty() && IsSidecar` (we need to use the appropriate function ofc)

Reply:
does the logic here not already match that? we're iterating directly over the `pod.Spec` which contains InitContainers and Containers (which includes both app and sidecar). 

The logic as-is also matches the logic used to assign static pods. If we change it here we would also need also modify https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cpumanager/policy_static.go#L480-L512

Reply:
sidecar containers are indeed implemented as init container, but per our convo in the issue we should disable quota for app and sidecar containers, aren't we?
Thing is: a sidecar container is a init container (plus the `RestartPolicy=Always`) but not all init containers are sidecar containers, this is why I'm advocating for further filtering here as I described above.

[The link ](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cpumanager/policy_static.go#L480-L512) LGTM already, we are making logic to distinguish pure init vs sidecar containers.